### ISSUE TITLE
ExternalMethod ZMI error in Zope 4.5.1 (Python 3.8.6)

### ISSUE THREAD

**Issue by georgpfolz** [zopefoundation/Products.ExternalMethod#12]
## BUG/PROBLEM REPORT

When trying to access an ExternalMethod in the ZMI, I get an internal server error.

**The ExternalMethod is fully functional when called, though.**

### What I did:

This happens with ExternalMethods created before the update to this Zope version as well as with newly created ones.

### What I expect to happen:

The ZMI dialogue

### What actually happened:

An internal server error:

```
2020-10-30 16:03:09 ERROR [Zope.SiteErrorLog:252][waitress] 1604070189.365760.9582454546275475 http://localhost:9643/get_hostname/manage_workspace
Traceback (innermost last):
  Module ZPublisher.WSGIPublisher, line 162, in transaction_pubevents
  Module ZPublisher.WSGIPublisher, line 359, in publish_module
  Module ZPublisher.WSGIPublisher, line 250, in publish
  Module ZPublisher.BaseRequest, line 644, in traverse
  Module ZPublisher.HTTPResponse, line 1031, in unauthorized
zExceptions.unauthorized.Unauthorized: You are not authorized to access this resource.
2020-10-30 16:03:09 ERROR [Zope.SiteErrorLog:252][waitress] 1604070189.3981740.467352058993902 http://localhost:9643/get_hostname/manage_main
Traceback (innermost last):
  Module ZPublisher.WSGIPublisher, line 162, in transaction_pubevents
  Module ZPublisher.WSGIPublisher, line 359, in publish_module
  Module ZPublisher.WSGIPublisher, line 254, in publish
  Module ZPublisher.mapply, line 85, in mapply
  Module ZPublisher.WSGIPublisher, line 63, in call_object
  Module Shared.DC.Scripts.Bindings, line 335, in __call__
  Module Shared.DC.Scripts.Bindings, line 372, in _bindAndExec
  Module App.special_dtml, line 213, in _exec
  Module DocumentTemplate._DocumentTemplate, line 145, in render_blocks
  Module DocumentTemplate._DocumentTemplate, line 167, in render_blocks_
  Module DocumentTemplate._DocumentTemplate, line 375, in __getitem__
  Module DocumentTemplate._DocumentTemplate, line 393, in getitem
  Module Shared.DC.Scripts.Bindings, line 349, in __render_with_namespace__
  Module Shared.DC.Scripts.Bindings, line 372, in _bindAndExec
  Module App.special_dtml, line 213, in _exec
  Module DocumentTemplate._DocumentTemplate, line 145, in render_blocks
  Module DocumentTemplate._DocumentTemplate, line 167, in render_blocks_
  Module DocumentTemplate._DocumentTemplate, line 375, in __getitem__
  Module DocumentTemplate._DocumentTemplate, line 393, in getitem
  Module Shared.DC.Scripts.Bindings, line 349, in __render_with_namespace__
  Module Shared.DC.Scripts.Bindings, line 372, in _bindAndExec
  Module App.special_dtml, line 213, in _exec
  Module DocumentTemplate._DocumentTemplate, line 145, in render_blocks
  Module DocumentTemplate._DocumentTemplate, line 237, in render_blocks_
  Module DocumentTemplate._DocumentTemplate, line 169, in render_blocks_
  Module DocumentTemplate.DT_Util, line 235, in eval
   - __traceback_info__: ZopeVersion
  Module <string>, line 1, in <module>
NameError: name 'ZopeVersion' is not defined
2020-10-30 16:03:09 ERROR [waitress:356][waitress] Exception while serving /get_hostname/manage_main
Traceback (most recent call last):
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/waitress/channel.py", line 349, in service
    task.service()
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/waitress/task.py", line 169, in service
    self.execute()
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/waitress/task.py", line 439, in execute
    app_iter = self.channel.server.application(environ, start_response)
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/ZPublisher/httpexceptions.py", line 30, in __call__
    return self.application(environ, start_response)
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/paste/translogger.py", line 69, in __call__
    return self.application(environ, replacement_start_response)
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/ZPublisher/WSGIPublisher.py", line 359, in publish_module
    response = _publish(request, new_mod_info)
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/ZPublisher/WSGIPublisher.py", line 254, in publish
    result = mapply(obj,
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/ZPublisher/mapply.py", line 85, in mapply
    return debug(object, args, context)
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/ZPublisher/WSGIPublisher.py", line 63, in call_object
    return obj(*args)
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/Shared/DC/Scripts/Bindings.py", line 335, in __call__
    return self._bindAndExec(args, kw, None)
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/Shared/DC/Scripts/Bindings.py", line 372, in _bindAndExec
    return self._exec(bound_data, args, kw)
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/App/special_dtml.py", line 213, in _exec
    result = render_blocks(self._v_blocks, ns,
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/DocumentTemplate/_DocumentTemplate.py", line 145, in render_blocks
    render_blocks_(blocks, rendered, md, encoding)
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/DocumentTemplate/_DocumentTemplate.py", line 167, in render_blocks_
    t = md[t]
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/DocumentTemplate/_DocumentTemplate.py", line 375, in __getitem__
    return self.getitem(name, call=1)
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/DocumentTemplate/_DocumentTemplate.py", line 393, in getitem
    return e.__render_with_namespace__(self)
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/Shared/DC/Scripts/Bindings.py", line 349, in __render_with_namespace__
    return self._bindAndExec((), namevals, namespace)
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/Shared/DC/Scripts/Bindings.py", line 372, in _bindAndExec
    return self._exec(bound_data, args, kw)
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/App/special_dtml.py", line 213, in _exec
    result = render_blocks(self._v_blocks, ns,
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/DocumentTemplate/_DocumentTemplate.py", line 145, in render_blocks
    render_blocks_(blocks, rendered, md, encoding)
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/DocumentTemplate/_DocumentTemplate.py", line 167, in render_blocks_
    t = md[t]
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/DocumentTemplate/_DocumentTemplate.py", line 375, in __getitem__
    return self.getitem(name, call=1)
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/DocumentTemplate/_DocumentTemplate.py", line 393, in getitem
    return e.__render_with_namespace__(self)
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/Shared/DC/Scripts/Bindings.py", line 349, in __render_with_namespace__
    return self._bindAndExec((), namevals, namespace)
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/Shared/DC/Scripts/Bindings.py", line 372, in _bindAndExec
    return self._exec(bound_data, args, kw)
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/App/special_dtml.py", line 213, in _exec
    result = render_blocks(self._v_blocks, ns,
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/DocumentTemplate/_DocumentTemplate.py", line 145, in render_blocks
    render_blocks_(blocks, rendered, md, encoding)
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/DocumentTemplate/_DocumentTemplate.py", line 237, in render_blocks_
    render_blocks_(block, rendered, md, encoding)
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/DocumentTemplate/_DocumentTemplate.py", line 169, in render_blocks_
    t = t(md)
  File "/Users/me/opt/myZope/lib/python3.8/site-packages/DocumentTemplate/DT_Util.py", line 235, in eval
    return eval(code, d)
  File "<string>", line 1, in <module>
NameError: name 'ZopeVersion' is not defined
```

### What version of Python and Zope/Addons I am using:

- Python 3.8.6
- Zope 4.5.1
- ExternalMethod 4.4

**Comment by jugmac00** [comment]
Possibly related with #9

**Comment by georgpfolz** [comment]
Sorry, my bad, I had ExternalMethods 4.3 installed with Zope 4.5.1. The error ist fixed in 4.4!
